py/torch_tensorrt/dynamo/runtime/_WrapperTorchTensorRTModule.py

```diff
@@ -0,0 +1,285 @@
+        compiled_module: Complied fx graphModule that will be wrapped
+        output_shapes: Shapes of output Tensors of the graph
+        output_dtypes: Output data types of the graph
+    Returns:
+        Output tensor or tensor list
+    """
+
+    def __init__(
+        self,
+        original_module: torch.nn.Module,
+        compiled_module: torch.nn.Module,
+        output_shapes: List[torch.Size],
+        output_dtypes: List[torch.dtype],
+    ):
+        super(WrapperTorchTensorRTModule, self).__init__()
+        self.original_module = original_module
+        self.compiled_module = compiled_module
+        self.inputs = partitioning.construct_submodule_inputs(compiled_module)
+        self.output_shapes = output_shapes
+        self.output_dtypes = output_dtypes
+
+        self._input_buffers: List[torch.Tensor] = []
+        self._output_buffers: List[torch.Tensor] = []
+        self.cudagraph: Optional[torch.cuda.CUDAGraph] = None
+        self.shape_key: Optional[str] = None
+        self.profiling_enabled = False
+        self.cudagraphs_enabled = False
+        self._caller_stream: Optional[torch.cuda.Stream] = None
+        self._engine_stream: Optional[torch.cuda.Stream] = None
+        self.input_is_dynamic = input_is_dynamic(self.inputs)
+
+        # Disable cudagrphs in submodules as it will be enabled in wrapper
+        for name, rt_mod in self.compiled_module.named_children():
+            if "_run_on_acc" in name:
+                rt_mod.set_whole_cudagraphs(True)
+
+        # Warm up is necessary to ensure that memory allocations and initializations are not recorded in cuda graphs
+        with unset_fake_temporarily():
+            inputs_tensor = [spec.torch_tensor.cuda() for spec in self.inputs]
+            s = torch.cuda.Stream()
+            s.wait_stream(torch.cuda.current_stream())
+            with torch.cuda.stream(s):
+                for _ in range(3):
+                    self.compiled_module(*inputs_tensor)
+            torch.cuda.current_stream().wait_stream(s)
+
+    def validate_input_shapes(self, inputs: Sequence[torch.Tensor]) -> bool:
+        """
+        Validates the input shapes of the forward function has changed
+        """
+        # Representation of input shapes to a given model
+        # Shapes are concatenated as so:
+        # x: (3, 4), y: (4, 5) --> Key: (3,4)(4,5)
+        new_shape_key = "".join(str(tuple(t.shape)).replace(" ", "") for t in inputs)
+
+        # If the new shape key differs from the existing one, infer new output shape
+        if new_shape_key != self.shape_key:
+            logger.debug(f"Input shape changed {self.shape_key} -> {new_shape_key}")
+            self.shape_key = new_shape_key
+
+            if self.input_is_dynamic:
+                with FakeTensorMode() as mode:
+                    fake_inputs = [mode.from_tensor(input) for input in inputs]
+                    tmp_outputs = self.original_module(*fake_inputs)
```

Comment:
Maybe I can use complied_module() instead. I'm now sure which one is better.

Reply:
compiled module is probably better